iceoryx_hoofs/source/posix_wrapper/shared_memory_object/shared_memory.cpp

```diff
@@ -142,7 +142,7 @@ cxx::expected<SharedMemory, SharedMemoryError> SharedMemoryBuilder::create() noe
                               << "\". This may be a SharedMemory leak." << std::endl;
                 });
 
```

Comment:
I think a good solution would be to add a check right before `ftrunctate` in line 122.
If here the shared memory `hasOwnership` and the `m_accessMode == AccessMode::READ_ONLY` it should fail.

For this you could introduce a new `SharedMemoryError` enum value and return it here in combination with an error message. This could then also be verified in the unit test and wouldn't fail in windows.

Or you could do a check right in the beginning of the `create()` method to verify that `OpenMode::**CREATE**` is not combined with `AccessMode::READ_ONLY`.

Reply:
@elfenpiff I implemented something similar to your suggestion. Checking before the `truncate` would work, but it could potentially hide errors. I believe we want to complain about READ_ONLY + OPEN_OR_CREATE, even if the file happens to already exist.

So I moved the check up, before the open-if-exist attempt. Since `hasOwnership` now has to be in scope during the open-check, I also used the opportunity to simplify the code there (the non-owning success path can now follow the same code path as the owning success path, with `hasOwnership` distinguishing between the two).

Pipeline seems to run throught, the invalid links are not caused by this MR.

/cc @mossmaurice